xpti/include/xpti/xpti_trace_framework.hpp

```diff
@@ -491,6 +491,18 @@ class scoped_notify {
                             m_user_data);
     }
   }
+  scoped_notify(uint8_t stream_id, uint16_t trace_type,
+                xpti::trace_event_data_t *parent,
+                xpti::trace_event_data_t *object, uint64_t instance,
+                const void *user_data = nullptr)
+      : m_object(object), m_parent(parent), m_stream_id(stream_id),
+        m_trace_type(trace_type), m_user_data(user_data), m_instance(instance) {
+    if (xptiTraceEnabled()) {
```

Comment:
nit: I'd like to see an early return instead.

Reply:
Done!